src/MultiIndices/FixedMultiIndexSet.cpp

```diff
@@ -204,24 +207,30 @@ FixedMultiIndexSet<MemorySpace>::FixedMultiIndexSet(unsigned int
 
 template<typename MemorySpace>
 FixedMultiIndexSet<MemorySpace>::FixedMultiIndexSet(unsigned int dim,
-                                                    unsigned int maxOrder) : dim(dim), isCompressed(true)
+                                                    unsigned int maxOrder,
+                                                    unsigned int minOrder) : dim(dim), isCompressed(true)
 {
     // Figure out the number of terms in the total order
     unsigned int numTerms, numNz;
     std::tie(numTerms,numNz) = TotalOrderSize(maxOrder, 0);
 
     // Allocate space for the multis in compressed form
     nzStarts = Kokkos::View<unsigned int*, MemorySpace>("nzStarts", numTerms+1);
-    nzDims   = Kokkos::View<unsigned int*, MemorySpace>("nzDims", numNz);
+    nzDims = Kokkos::View<unsigned int*, MemorySpace>("nzDims", numNz);
     nzOrders = Kokkos::View<unsigned int*, MemorySpace>("nzOrders", numNz);
 
-    // Compute the multis
-    std::vector<unsigned int> workspace(dim);
-    unsigned int currNz=0;
-    unsigned int currTerm=0;
-
-    FillTotalOrder(maxOrder, workspace, 0, currTerm, currNz);
-
+    // Put in a parallel for loop of length 1 to make sure it's executed on device
+    Kokkos::View<unsigned int*, MemorySpace> workspace("workspace", dim);
+    unsigned int newNumTerms = 0;
+    Kokkos::parallel_reduce("Dummy Loop", 1, KOKKOS_LAMBDA (const int blahind, unsigned int& currTerm) {
+        unsigned int currNz=0;
+        FillTotalOrder(maxOrder, minOrder, workspace, 0, currTerm, currNz);
+    }, newNumTerms);
+    
+    Kokkos::resize(nzStarts, newNumTerms+1);
```

Comment:
Instead of figuring out exactly how many terms will be needed for a given `maxOrder` and `minOrder`, I'm initially allocating space as if `minOrder=0`.  These resizes are to then shrink the `Kokkos::View` to only what's needed.  Based on the Kokkos documentation, this seems like it will work on both device and host arrays.